cmd/algorelay/relayCmd.go

```diff
@@ -380,26 +380,43 @@ func ensureRelayStatus(checkOnly bool, relay eb.Relay, nameDomain string, srvDom
 	}
 
 	targetDomainAlias := relay.DNSAlias + "." + nameDomain
+
 	if topmost != targetDomainAlias {
 		if checkOnly {
 			err = fmt.Errorf("topmost DNS name is not the assigned DNS Alias (wanted: %s, found %s)",
 				relay.DNSAlias, topmost)
 			return
 		}
 
-		// Add A/CNAME for the DNSAlias assigned
-		err = addDNSRecord(targetDomainAlias, topmost, ctx.nameZoneID)
+		if relay.CheckSuccess {
+			// Add A/CNAME for the DNSAlias assigned
+			err = addDNSRecord(targetDomainAlias, topmost, ctx.nameZoneID)
+			if err != nil {
+				return
+			}
+			fmt.Printf("[%d] Added DNS Record: %s -> %s\n", relay.ID, targetDomainAlias, topmost)
+
+			// Update our state
+			names = append(names, targetDomainAlias)
+			topmost = targetDomainAlias
+		} else {
+			// remove entry.
+			err = deleteDNSRecord(targetDomainAlias, topmost, ctx.nameZoneID)
+			if err != nil {
+				return
+			}
+			fmt.Printf("[%d] Removed DNS Record: %s -> %s\n", relay.ID, targetDomainAlias, topmost)
+		}
+	} else if !relay.CheckSuccess {
+		// remove entry.
```

Comment:
In the other branch (topmost != targetDomainAlias), it is deleted only if checkOnly is false. 
Why checkOnly is not considered here?

Reply:
if you'll look carefully, you'll see that `ensureRelayStatus` is called with `checkOnly=true` only when `relay.CheckSuccess=true`. ( see `checkCmd` and `updateCmd`